File: Python/Naive_String_matching.py
### T - largrt text ### P - pattern to match

def BruteForceMatch(T,P):
	i = 0			
	for i in range(len(T)-len(P)+1):	
		j = 0
		for j in range(len(P)):
			if T[i+j] != P[j]:			
				break
			else:
				if j == len(P)-1:
					print "match is found at" , i+1 ,"to ", i+j+1 ,"(letter no of the text)"
					i += j
			

txt = raw_input("Enter the larger text: ")
ptrn = raw_input("Enter the pattern want to match: ")
BruteForceMatch(txt,ptrn)

File: cpp/Naive_String_matching.cpp
#include <iostream>
#include <string.h>


int BruteForceMatch(std::string,std::string);

int main()
{
	std::string txt,ptrn;
	std::cout<<"Enter the Text: ";
	std::cin>>txt;
	std::cout<<"Enter the pattern want to match: ";
	std::cin>>ptrn;
	BruteForceMatch(txt,ptrn);
	
	return 0;
}

int BruteForceMatch(std::string txt,std::string ptrn)
{
	int i = 0,j,txtlen,ptrnlen;
	txtlen = txt.length();
	ptrnlen = ptrn.length();
	
	while (i < txtlen)
	{
		j = 0;
		while (j < ptrnlen)
		{
			if (txt[i+j] != ptrn[j])
			{
				break;
			}
			else
			{
				if ((txt[i+j] == ptrn[j]) && (j == (ptrnlen-1)))
				{
					std::cout<<"match found from "<<i+1<<" to "<<i+j+1<<'\n'; 
					i += j+1;
					j = 0;
				}
			}
			j++;
		}
		i++;
	}
	return 0;
}
